app/src/main/java/com/github/fribourgsdp/radio/UserProfileActivity.kt

```diff
@@ -156,12 +152,20 @@ class UserProfileActivity : AppCompatActivity() {
                     user.save(this)
                 }
                 usernameField.setText(user.name)
-                googleSignInButton.setText(getString(R.string.sign_out_message))
+                googleSignInButton.text = getString(R.string.sign_out_message)
 
                 signedIn = true
             }
         }
     }
+
+    override fun onBackPressed() {
+        super.onBackPressed()
+        val intent = Intent(this, MainActivity::class.java)
+        intent.putExtra(RECREATE_USER, false)
+        startActivity(intent)
+        finish()
+    }
```

Comment:
Really not a fan of this as it can really mess with fragments when we don't intend for back to return to main menu. Is it still necessary? if we remove it, can we not just never create a user and create it in the try / catch of no use being found on load?